packages/block-library/src/file/block.json

```diff
@@ -57,6 +57,17 @@
 	"supports": {
 		"anchor": true,
 		"align": true,
+		"spacing": {
+			"margin": true,
+			"padding": true
+		},
+		"__experimentalStyle": {
+			"spacing": {
+				"margin": {
+					"bottom": "1.5em"
+				}
+			}
+		},
```

Comment:
Unfortunately, I don't think we can make this switch yet as it would be a regression for themes without a theme.json file due to the generated styles not being loaded for them.

See: https://github.com/WordPress/gutenberg/issues/46818

Reply:
Thanks for mentioning this, and for linking through to the WIP work to fix layout conflicts with margins. Something that's also relevant here is that in the exploration by @tellthemachines over in https://github.com/WordPress/gutenberg/pull/47858 the Columns block's default `margin-bottom` rule has its specificity reduced. The idea in that PR is that for layout rules to be able to have reduced specificity, we likely will also need all core blocks' default margin rules to be lower specificity too (e.g. wrapped in a `:where()` rule) so that low specificity layout rules can easily override it.

Reply:
> Unfortunately, I don't think we can make this switch yet as it would be a regression for themes without a theme.json file due to the generated styles not being loaded for them.

There is a specific stylesheet loaded for them now though? The style should be added there.

Reply:
I wasn't aware there was. I was basing my comment off both https://github.com/WordPress/gutenberg/issues/46818 and https://github.com/WordPress/gutenberg/issues/45198 which are still open indicating the issue remains.

Reply:
Here is where the classic.scss file was introduced.
https://github.com/WordPress/gutenberg/pull/44334

https://developer.wordpress.org/reference/functions/wp_enqueue_classic_theme_styles/
In 6.2 it uses `wp_theme_has_theme_json() ` to determine when the file is loaded.

Reply:
Thank you for trying out the classic.scss approach @carolinan

Unfortunately, I think things have changed a bit recently. https://github.com/WordPress/gutenberg/pull/47858 landed which fixed the layout margin rule specificity. Part of that meant lowering the specificity of some blocks' CSS rules. This included the File block.

The current state of this PR means we would have the margin rule in 3 places (style.scss, classic.scss, and block.json). 

I suspect we might be better off removing the addition of the new rule to `classic.scss` and the `block.json` via `__experimentalStyle`.  The latter overrides the lower margin specificity required to work alongside layout supports.

Reply:
I don't think the layout margin rules are fixed, there are issues with two of the other blocks I have spacing PR's for.

Reply:
The "fixing" of margin rules was so that a block's global styles for margin could be applied. I believe the fix from https://github.com/WordPress/gutenberg/pull/47858 is still working. It sounds like the interaction between the overall layout styles and top-level block margins is what you don't think is working correctly?

I'm about to start looking more closely at the [Preformatted spacing PR](https://github.com/WordPress/gutenberg/pull/45196). 

The left/right margins not being applied for top-level blocks makes sense but I need to have a play around with the PR to see what's happening on the frontend. I'll comment further on that PR.

Reply:
Coming back to this one with fresh eyes, I still think we just need to remove the classic.scss style and this `__experimentalStyle` entry and we should be able to move ahead with this PR.

Reply:
Removed 👍